packages/eslint-plugin/lib/rules/polaris-modal-requires-activator-prop.js

```diff
@@ -0,0 +1,34 @@
+const {docsUrl, polarisComponentFromJSX} = require('../utilities');
+
+module.exports = {
+  meta: {
+    docs: {
+      description:
+        'Polaris <Modal /> and components that extend it must provide an `activator` prop to ensure proper keyboard focus handling on close',
+      category: 'Best Practices',
+      recommended: true,
+      uri: docsUrl('polaris-modal-requires-activator-prop'),
+    },
+    schema: [],
+  },
+
+  create(context) {
+    return {
+      JSXElement(node) {
+        const component = polarisComponentFromJSX(node, context);
+        if (
+          component === 'Modal' &&
+          node.openingElement.attributes.filter(
+            (attribute) => attribute.name.name === 'activator',
```

Comment:
The two issues (2020, 2021) where a linter was proposed as a way forward both say that making `activator` required would be a breaking change in Polaris and thus is a non-starter.

Reply:
I understand it would be a breaking change to introduce in Polaris right now or back in 2020-2021, I was referring to the fact that it was implemented there initially with an activator being optional before being used by consumers, so they _might_ be able to provide us the reasoning behind making it optional as opposed to required in the first place.

Reply:
Ahhhh got it. Also got some perspective from Ben on the Polaris release cycle. I know someone currently on that team so I can do a little investigation! Thanks for clarifying patiently :)